File: backend/app/schemas/analytics.py
from decimal import Decimal

from pydantic import BaseModel


class ProductSalesSummary(BaseModel):
    product_name: str
    quantity: int
    revenue: Decimal


class HourSalesSummary(BaseModel):
    hour: int
    orders: int
    revenue: Decimal


class BreakdownSummary(BaseModel):
    label: str
    count: int


class DashboardAnalyticsResponse(BaseModel):
    revenue: Decimal
    order_count: int
    average_ticket: Decimal
    top_product: ProductSalesSummary | None
    products: list[ProductSalesSummary]
    sales_by_hour: list[HourSalesSummary]
    orders_by_status: list[BreakdownSummary]
    orders_by_payment: list[BreakdownSummary]

File: frontend/src/services/analytics.ts
import { api } from './api'

export type DashboardAnalytics = {
  revenue: string
  order_count: number
  average_ticket: string
  top_product: { product_name: string; quantity: number; revenue: string } | null
  products: Array<{ product_name: string; quantity: number; revenue: string }>
  sales_by_hour: Array<{ hour: number; orders: number; revenue: string }>
  orders_by_status: Array<{ label: string; count: number }>
  orders_by_payment: Array<{ label: string; count: number }>
}

export async function getDashboardAnalytics() {
  const { data } = await api.get<DashboardAnalytics>('/analytics/dashboard')
  return data
}
